symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php

```diff
@@ -23,43 +23,51 @@
  * Description of GoogleAuthProvider
  */
 class GoogleAuthProvider extends AbstractAuthProvider {
-
     const PROFILE_SCOPE = 'profile';
     const EMAIL_SCOPE = 'email';
-
-    public function validateUser($provider, $authProvider = null) {
-
+    protected $loginService = null;
+    protected $option = array();
+    protected $authenticationMassage = '';
+    /**
+     * @param OpenidProvider $provider
+     * @param AuthProviderExtraDetails $authProvider
+     * @return array
+     */
+    public function validateUser($provider, $authProvider = null)
+    {
         $gClient = new Google_Client();
         $gClient->setApplicationName('Login to Google +');
         $gClient->setClientId($authProvider->getClientId());
         $gClient->setClientSecret($authProvider->getClientSecret());
         $gClient->setRedirectUri($provider->getProviderUrl());
         $gClient->setDeveloperKey($authProvider->getDeveloperKey());
-        $gClient->addScope(array(self::EMAIL_SCOPE,self::PROFILE_SCOPE));
-
-        $gService = new Google_Service_Plus($gClient);
+        $gClient->addScope(array(self::EMAIL_SCOPE, self::PROFILE_SCOPE));
+        $requestParameters = $this->getOption();
+        $requestCode = $requestParameters['code'];
 
-        if (isset($_GET['code'])) {
-            $gClient->authenticate($_GET['code']);
+        if (isset($requestCode)) {
```

Comment:
This is not right .. it shild be like this
if($requestParameters['code']) { 
...
}